blocks/subscriptions-block/features/checkout/_children/Review/index.jsx

```diff
@@ -0,0 +1,394 @@
+import React, { useEffect, useState } from "react";
+import { useElements, useStripe } from "@stripe/react-stripe-js";
+
+import {
+	Paragraph,
+	usePhrases,
+	Stack,
+	Button,
+	useIdentity,
+	useSales,
+} from "@wpmedia/arc-themes-components";
+
+import OrderInformation from "../../../../components/OrderInformation";
+import RenewalInformation from "../../../../components/RenewalInformation";
+import PaymentIcon from "../../../../components/PaymentIcons";
+
+import {
+	STRIPEINTENTS,
+	PAYPAL,
+	APPLEPAY,
+	GOOGLEPAY,
+	ARCXP_ORDERNUMBER,
+	RECAPTCHA_TOKEN,
+} from "../../../../utils/constants";
+
+export const StripeIntentsButtons = ({
+	paymentOptionSelected,
+	paymentMethodAppleGooglePay,
+	handleSubmitStripeIntents,
+	handleSubmitApplePayGooglePay,
+	isSubmitting,
+}) => {
+	const phrases = usePhrases();
+
+	const elements = useElements();
+	const stripe = useStripe();
+
+	/* istanbul ignore next */
+	const mountApplePayGooglePayButton = () => {
+		if (paymentMethodAppleGooglePay) {
+			const prButton = elements.create("paymentRequestButton", {
+				paymentRequest: paymentMethodAppleGooglePay,
+			});
+
+			prButton.mount("#ApplePay-payment-request-button");
+
+			paymentMethodAppleGooglePay.on("paymentmethod", async (e) => {
+				handleSubmitApplePayGooglePay(e, stripe);
+			});
+		}
+	};
+
+	/* istanbul ignore next */
+	useEffect(() => {
+		if (paymentOptionSelected === APPLEPAY || paymentOptionSelected === GOOGLEPAY) {
+			mountApplePayGooglePayButton();
+		}
+		// eslint-disable-next-line
+	}, [paymentOptionSelected, APPLEPAY, GOOGLEPAY]);
```

Comment:
This is another case where the lint is trying to tell you something.

`mountApplePayGooglePayButton` is only ever used in this useEffect, so it's not necessary to create it evert render on line 39.

It should probably be something like this:
```suggestion
	useEffect(() => {
		if (paymentOptionSelected === APPLEPAY || paymentOptionSelected === GOOGLEPAY) {
			if (paymentMethodAppleGooglePay) {
				const prButton = elements.create("paymentRequestButton", {
					paymentRequest: paymentMethodAppleGooglePay,
				});

				prButton.mount("#ApplePay-payment-request-button");

				paymentMethodAppleGooglePay.on("paymentmethod", async (e) => {
					handleSubmitApplePayGooglePay(e, stripe);
				});
			}
		}
	}, [elements, paymentMethodAppleGooglePay, paymentOptionSelected, APPLEPAY, GOOGLEPAY]);
```
The other thing I worry about here is the `.on("paymentmethod"` call seems to be adding an event, but the event is never cleaned up?  Should there be a event removal cleanup on this effect?

Reply:
@nschubach attending this feedback I found an issue which was fixed. On `<Payment />` We need to check if the element exists, if it exists, the element should be destroyed ([stripe Intent documentation](https://docs.stripe.com/js/element/other_methods/destroy)), and on `<Review />` we need to create the element again (button that handles the ApplePay/GooglePay via stripe intents.). We don't need to worry about `paymentMethodAppleGooglePay.on()`, since on <Review  /> the user should complete the payment, and if it was a success, the user should be redirected to another page.

Reply:
👍